pkg/controller/job/job_controller.go

```diff
@@ -1693,11 +1696,32 @@ func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType
 	return nil
 }
 
-func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus) {
+func recordJobPodFinished(job, oldJob *batch.Job, newSucceededIndexes []int) {
 	completionMode := completionModeStr(job)
-	diff := job.Status.Succeeded - oldCounters.Succeeded
+	var diff int
+
+	// Updating succeeded metric must be handled differently
+	// for Indexed Jobs to handle the case where the job has
+	// been scaled down by reducing completions & parallelism
+	// in tandem, and now a previously completed index is
+	// now out of range (i.e. index >= spec.Completions).
+	if isIndexedJob(job) {
+		prevUniqueIndexes := succeededIndexesFromJob(oldJob, false)
```

Comment:
I'm not convinced that passing `(oldJob, true)` here makes any difference, because `newSucceededIndexes` doesn't include any indexes past `completions`.

Reply:
> Did you try this? succeededIndexes.total() - prevSucceededIndexes.total()

Yes, it results in the test case `indexed job past and new pods` failing due to `recordJobPodFinished()` being called twice during the test, so simply passing in this delta results in double counting.

Reply:
> I'm not convinced that passing (oldJob, true) here makes any difference, because newSucceededIndexes doesn't include any indexes past completions.

Looks like you're right! I have updated the code accordingly so `succeededIndexesFromJob` is in its original state, as there was nowhere that I needed to set `filterOutOfRangeIndexes` to false.